src/pages/iou/request/step/IOURequestStepCategory.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import CategoryPicker from '@components/CategoryPicker';
+import type {ListItem} from '@components/SelectionList/types';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as TransactionUtils from '@libs/TransactionUtils';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Policy, PolicyCategories, PolicyTagList, ReportActions, Session, Transaction} from '@src/types/onyx';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import type {WithWritableReportOrNotFoundProps} from './withWritableReportOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+type IOURequestStepCategoryOnyxProps = {
+    /** The draft transaction that holds data to be persisted on the current transaction */
+    splitDraftTransaction: OnyxEntry<Transaction>;
+
+    /** The policy of the report */
+    policy: OnyxEntry<Policy>;
+
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<PolicyCategories>;
+
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxEntry<PolicyTagList>;
+
+    /** The actions from the parent report */
+    reportActions: OnyxEntry<ReportActions>;
+
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+type IOURequestStepCategoryProps = IOURequestStepCategoryOnyxProps &
+    WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_CATEGORY> & {
+        /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+        transaction: OnyxEntry<Transaction>;
+    };
+
+function IOURequestStepCategory({
+    report,
+    route: {
+        params: {transactionID, backTo, action, iouType, reportActionID},
+    },
+    transaction,
+    splitDraftTransaction,
+    policy,
+    policyTags,
+    policyCategories,
+    reportActions,
+    session,
+}: IOURequestStepCategoryProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const isEditing = action === CONST.IOU.ACTION.EDIT;
+    const isEditingSplitBill = isEditing && iouType === CONST.IOU.TYPE.SPLIT;
+    const transactionCategory = ReportUtils.getTransactionDetails(isEditingSplitBill && !lodashIsEmpty(splitDraftTransaction) ? splitDraftTransaction : transaction)?.category;
+
+    const reportAction = reportActions?.[report?.parentReportActionID ?? reportActionID] ?? null;
```

Comment:
Just saying: ?? instead of || might change the logic a bit, you might have to use eslint disable comment.

Reply:
An in other places too, just be careful and let's test the component thoroughly

Reply:
it didn't seem to break this, but I don't want to risk breaking anything, I'll leave it as it was